embedded/devices/src/ads1219.cpp

```diff
@@ -0,0 +1,64 @@
+#include "embedded/devices/include/ads1219.h"
+
+
+#define WRITE_CONFIG_REG 0x40 // 0100 00xx
+#define READ_CONFIG_REG 0x20 // 0010 00xx
+#define READ_STATUS_REG 0x24 // 0010 01xx
+#define RESET 0x06 // 0000 011x
+#define RDATA 0x10 // 0001 xxxx
+#define START_CONVERSION 0x08 // 0000 100x
+#define CONVERSION_VOLTAGE 244.14e-9
+
+Ads1219::Ads1219(int bus, int addr) : I2c(bus, addr, O_RDWR){}
+
+void Ads1219::loop() {
+    while (this->stop_thread != true) {
+        // get the current channel we are on
+        uint8_t configReg = this->read_from_reg(READ_CONFIG_REG); 
+        uint8_t currentChannel = configReg >> 5;
+        // calculate the next channel
+        if(currentChannel == 6 || currentChannel == 0){
```

Comment:
I think we should be checking if `currentChannel == 7`, not 6. We want to cover the cases where this part of the register is:

011 for channel 0
100 for channel 1
101 for channel 2
110 for channel 3

If we want to read channel 3, I think we need to check if the current val of the register == 7

Reply:
this isn't how it works. currentChannel is the channel that we just checked in this case